File: examples/benchmark.c
#include <math.h>

int isPrime(int n) {
    if (n < 2)
        return 0;
    else if (n == 2)
        return 1;
    else if (n % 2 == 0)
        return 0;

    int upperLimit = sqrt(n);
    int i = 3;
    while (i <= upperLimit) {
        if (n % i == 0)
            return 0;
        i += 2;
    }
    return 1;
}

int main() {
    int i;
    for (i = 1; i < 2000000; i++) {
        isPrime(i);
    }
    return 0;
}

File: examples/benchmark.py
#!/usr/bin/env python
from math import sqrt

def isPrime(n):
    if (n < 2): return False
    elif (n == 2): return True
    elif (n % 2 == 0): return False

    upper_limit = int(sqrt(n))
    i = 3
    while (i <= upper_limit):
        if (n % i == 0): return False
        i += 2

    return True

lim = 2000000
no_primes = 0
n = 0

while (n <= lim):
    if (isPrime(n)): no_primes += 1
    n += 1

print ("Number of primes in the interval [0,%d]: %d" % (lim, no_primes))
